classes/note/Repository.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * @file classes/note/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @see Note
+ *
+ * @brief Operations for retrieving and modifying Note objects.
+ */
+
+namespace PKP\note;
+
+use Carbon\Carbon;
+
+class Repository
+{
+    /**
+     * Fetch a note by symbolic info, building it if needed.
+     */
+    public function build(int $assocType, int $assocId, ?int $userId, ?string $contents, ?string $title): Note
+    {
+        return Note::withUserId($userId)
+            ->withAssoc($assocType, $assocId)
+            ->firstOr(fn() => Note::create([
+                'assocType' => $assocType,
+                'assocId' => $assocId,
+                'userId' => $userId,
+                'contents' => $contents,
+                'title' => $title,
+                'dateCreated' => Carbon::now(),
+                'dateModified' => Carbon::now(),
+            ]));
+    }
+}
+
+if (!PKP_STRICT_MODE) {
+    class_alias('\PKP\note\Note', '\Note');
+    define('NOTE_ORDER_DATE_CREATED', Note::NOTE_ORDER_DATE_CREATED);
+    define('NOTE_ORDER_ID', Note::NOTE_ORDER_ID);
+}
```

Comment:
don't think this should be here

Reply:
In fact, since we're breaking backwards compatibility, now is a good time to remove the `define` / `class_alias` calls entirely. The remaining ones will go away with 3.6, but we're removing them opportunistically when we introduce a breaking change, like shifting over to Eloquent.